fs/reader/reader.go

```diff
@@ -201,13 +203,18 @@ func (gr *reader) isClosed() bool {
 }
 
 type file struct {
-	id uint32
-	fr metadata.File
-	gr *reader
+	id       uint32
+	fr       metadata.File
+	gr       *reader
+	verified atomic.Bool
+	lock     sync.Mutex
 }
 
 // ReadAt reads the file when the file is requested by the container
 func (sf *file) ReadAt(p []byte, offset int64) (int, error) {
+	if err := sf.Verify(); err != nil {
```

Comment:
Given this will have some impact on the runtime, does it make sense to have a config option to disable file verification? Something like `DisableFileVerfication`?

Reply:
Yeah, we inherited a `disable_verification` bool that we're ignoring right now. It was originally meant for validating the estargz TOC against a label passed when launching the container, but that doesn't really make sense for SOCI.

I think we can probably clean that up and reuse that field. I will create an issue for that.

Reply:
https://github.com/awslabs/soci-snapshotter/issues/819